File: server/serve/src/main/java/com/qqsystem/serve/entity/Actor.java
package com.qqsystem.serve.entity;

import com.fasterxml.jackson.annotation.JsonFormat;
import lombok.Data;

import java.time.LocalDateTime;
import java.util.List;

@Data
public class Actor {

    private Long id;

    private String name;

    private String avatar;

    private String intro;

    private String roleName;

    private String style;

    private Integer worksCount;

    private Integer viewCount;

    private Integer likeCount;

    @JsonFormat(pattern = "yyyy-MM-dd HH:mm:ss", timezone = "Asia/Shanghai")
    private LocalDateTime joinDate;

    private Integer status;

    @JsonFormat(pattern = "yyyy-MM-dd HH:mm:ss", timezone = "Asia/Shanghai")
    private LocalDateTime createTime;

    @JsonFormat(pattern = "yyyy-MM-dd HH:mm:ss", timezone = "Asia/Shanghai")
    private LocalDateTime updateTime;

    // 关联扩展
    private List<Drama> dramas;
}

File: client/src/api/actor.ts
import request from './request'

export interface Actor {
  id: number
  name: string
  title: string
  avatar: string
  works: string[]
  fansCount: number
  worksCount: number
  awardsCount: number
}

export interface ActorDetail {
  id: number
  name: string
  title: string
  avatar: string
  intro: string
  roleName: string
  style: string
  worksCount: number
  viewCount: number
  likeCount: number
  joinDate: string
  status: number
  createTime: string
  updateTime: string
  dramas: any[]
}

export interface ActorListResponse {
  list: Actor[]
  total: number
}

export const getActorList = (page: number, size: number, style?: string) => {
  const params: any = {
    page,
    size
  }
  if (style !== undefined && style !== '') {
    params.style = style
  }
  return request<ActorListResponse>({
    url: '/actor/list',
    method: 'get',
    params
  })
}

export const getActorDetail = (id: number) => {
  return request<ActorDetail>({
    url: `/actor/detail/full/${id}`,
    method: 'get'
  })
}
